src/main/java/teammates/ui/controller/InstructorCourseInstructorDeleteAction.java

```diff
@@ -27,6 +28,17 @@ protected ActionResult execute() {
 
         /* Process deleting an instructor and setup status to be shown to user and admin */
         if (hasAlternativeInstructor(courseId, instructorEmail)) {
+            try {
+                validateIsInstructorDisplaytoStudents(courseId, instructorEmail);
+            } catch (InvalidParametersException e) {
+                isError = true;
+                setStatusForException(e);
```

Comment:
I think you're creating unnecessary complexity here, again because your `validate` method returns string. Why not just:
```java
List<InstructorAttributes> instructors = logic....;
isError = !logic.isInstructor...(instructors, instructorEmail);
if (isError) {
    setStatusForException(...);
    RedirectResult result = ...;
    result.addResponseParam(...);
    return result;
}
```
This way, it's more concise and you don't need an Exception (whose stack trace got lost anyway).